File: 101-200/108-convert-sorted-array-to-binary-search-tree.cpp
TreeNode* sortedArrayToBST(vector<int>& nums) {
    return helper(0, nums.size(), nums);
}

TreeNode* helper(int start, int after, vector<int>& nums) {
    if (start == after) return nullptr;
    int mid = start + (after-start)/2;
    TreeNode* root = new TreeNode(nums[mid]);
    root->left = helper(start, mid, nums);
    root->right = helper(mid+1, after, nums);
    return root;
}

File: python version/108-convert-sorted-array-to-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        n = len(nums)
        if n == 0: return None
        root = TreeNode(nums[n//2])
        root.left = self.sortedArrayToBST(nums[0:n//2])
        root.right = self.sortedArrayToBST(nums[n//2+1:n])
        return root